trie/trie.go

```diff
@@ -667,10 +637,6 @@ func (t *Trie) Reset() {
 	t.root = nil
 	t.owner = common.Hash{}
 	t.unhashed = 0
+	t.db = nil
```

Comment:
This `Reset` operation is interesting. Previously it was a bit strange, unsure what it could be used for. However, now it seems even less useful since it nukes out the db, so it gets pretty dead. There also doesn't seem to be any code that actually calls this method, so I guess we should just drop the entire thing?

Reply:
Yes, previously trie is used for deriving hash of txs and recepits. But we already switch to stacktrie. I think we can delete it.

Reply:
hmm, we still use it for testing

```go
func TestDeriveSha(t *testing.T) {
	txs, err := genTxs(0)
	if err != nil {
		t.Fatal(err)
	}
	for len(txs) < 1000 {
		exp := types.DeriveSha(txs, trie.NewEmpty(trie.NewDatabase(rawdb.NewMemoryDatabase())))
		got := types.DeriveSha(txs, trie.NewStackTrie(nil))
		if !bytes.Equal(got[:], exp[:]) {
			t.Fatalf("%d txs: got %x exp %x", len(txs), got, exp)
		}
		newTxs, err := genTxs(uint64(len(txs) + 1))
		if err != nil {
			t.Fatal(err)
		}
		txs = append(txs, newTxs...)
	}
}
```

Should we still keep it?

Reply:
Unsure, we can keep it, but then does setting the db to nil is necessary? Is that not the trie.Database?

Reply:
It's trie.Database.. I will revert it.

Reply:
done